I created my brunch Delivery to Functionality

shibiza opened the pull request:
…s. All 3 tests are passed

shibiza committed:
create my brunch @Delivery to Functionality@ and there java test class. All 3 tests are passed

mat-kud commented:
Use .gitignore to upload only necessary files (.idea folder is not necessary). Get rid of repetitive code by using @BeforeTest or @BeforeMethod. Try to write comments in English. Line 25 redundant assertion.

sabinawozniak reviewed (changes requested):
your comments are not readable. could you rewrite them?
I know that "var" can be "everything", but I don't know how exactly it works - if you are sure that var correctly recognized the type of variable, that's ok - you have only one variable, so I think that this may not be a problem. What is more , it is a feature in Java 10 and I don't know if we are able to use it, but on the other side - why not? :)

shibiza commented:
> Use .gitignore to upload only necessary files (.idea folder is not necessary). Get rid of repetitive code by using @BeforeTest or @BeforeMethod. Try to write comments in English. Line 25 redundant assertion.

I'm using this link to prepare gitignore:
https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore


this is a perfect example of gitignore and I strongly recommend you to use it!

shibiza commented:
> your comments are not readable. could you rewrite them? I know that "var" can be "everything", but I don't know how exactly it works - if you are sure that var correctly recognized the type of variable, that's ok - you have only one variable, so I think that this may not be a problem. What is more , it is a feature in Java 10 and I don't know if we are able to use it, but on the other side - why not? :)

thank you! I wrote them in english! soon will commit changes

shibiza committed:
created @BeforeMethod and @AfterMethod annotations.
changed "var" to String.

shibiza commented:
> Use .gitignore to upload only necessary files (.idea folder is not necessary). Get rid of repetitive code by using @BeforeTest or @BeforeMethod. Try to write comments in English. Line 25 redundant assertion.

so I deleted line #25, it was unnessessary. Comments are in English. Now I added annotations @BeforeTest and @AfterTest. My code is nice now! Thanx!

shibiza commented:
> your comments are not readable. could you rewrite them? I know that "var" can be "everything", but I don't know how exactly it works - if you are sure that var correctly recognized the type of variable, that's ok - you have only one variable, so I think that this may not be a problem. What is more , it is a feature in Java 10 and I don't know if we are able to use it, but on the other side - why not? :)

comments are in English (were in Russian). And instead of "var" I'm using "String". Thank you!

sabinawozniak reviewed (changes requested):
You can delete "AppTest" - we don't need it.
You use absolute path to chromedriver.exe - Yuliia said that we should use dependency from previous lesson.

shibiza committed:
deleted AppTest.java

shibiza committed:
got rid of the absolute path to chromedriver.exe (changed pom.xml ant setup for WebDriver)

sabinawozniak commented on src/test/java/org/example/DeliveryToFunctionality.java:
```diff
@@ -21,7 +22,8 @@ public class DeliveryToFunctionality {
 
```
Before @Before Method you should set up for example: "private WebDriver webdriver" - without it this code doesn't work, because it doesn't recognize "webdriver".

sabinawozniak reviewed (changes requested):
You use e.g."zipCode.sendKeys(zip);" and "...equals(poland);", but I don't see where you set up these variables, or (if it have to be string) you should add "".

mat-kud commented:
You didnt delete unnecessary folders (.idea, target) from your branch. Try to make your own .gitignore file instead of using generic one

shibiza committed:
It's not required but I've changed gitignore

shibiza committed:
I've changed gitignore

Petr-Kv commented on src/test/java/org/example/DeliveryToFunctionality.java:
```diff
@@ -0,0 +1,127 @@
+    @BeforeMethod
+    public void setup() {
+        WebDriverManager.chromedriver().setup();
+        // System.setProperty("webdriver.chrome.driver", "E:\\it\\java\\web_driver_test\\src\\test\\resources\\webdriver\\chromedriver.exe");
```
Why did you leave this comment

shibiza replied:
This comment I leave for myself)))) just to have an example how it could be. Don't pay attention to it

juliaHrabovska replied:
Yes, it is better to remove it. On real projects, we do not do like this

mat-kud commented:
.idea folder is still present in your branch

shibiza committed:
I've deleted idea folder

juliaHrabovska commented on src/test/java/org/example/DeliveryToFunctionality.java:
```diff
@@ -0,0 +1,127 @@
+        boolean waitToReload = new WebDriverWait(webDriver, Duration.ofSeconds(10)).until(ExpectedConditions.stalenessOf(destination));
+
+        WebElement zipCodeInfo = new WebDriverWait(webDriver, Duration.ofSeconds(10))
+                .until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//span[@id= 'nav-global-location-data-modal-action']//span[contains(@id, 'glow-ingress-line2')]")));
```
Not critical, but it is better to make this line shorter. It will be more readable.
if you have an element with id, you can just get this element. I mean, you can make this xpath shorter like this: //span[contains(@id, 'glow-ingress-line2')]

juliaHrabovska replied:
Same for other lines
